### PR TITLE
Add set_seed API

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Expose set_seed API from onxnruntime to torch_ort module

### CODE REVIEW

**Review by baijumeswani — APPROVED** [review]

File: tests/torch-ort_test_api.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+class NeuralNetSinglePositionalArgument(torch.nn.Module):
+    def __init__(self, input_size, hidden_size, num_classes):
+        super(NeuralNetSinglePositionalArgument, self).__init__()
+
+        self.fc1 = torch.nn.Linear(input_size, hidden_size)
+        self.relu = torch.nn.ReLU()
+        self.fc2 = torch.nn.Linear(hidden_size, num_classes)
+
+    def forward(self, input1):
```
Comment (SherlockNoMad) [inline tests/torch-ort_test_api.py:23]: There is no randomness in this model, let's add a dropout layer.

File: tests/torch-ort_test_api.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        out = self.fc1(input1)
+        out = self.relu(out)
+        out = self.fc2(out)
+        return out
+
+def test_set_seed():
+    device = 'cuda'
+    N, D_in, H, D_out = 64, 784, 500, 10
+    input = torch.randn(N, D_in, device=device)
```
Comment (SherlockNoMad) [inline tests/torch-ort_test_api.py:32]: the input and initial weight's randomness are both controlled by torch's seed. 
to test the randomness in ORT, we need a dropout layer.

File: tests/torch-ort_test_api.py (hunk 1 above)
Comment (thiagocrepaldi) [reply tests/torch-ort_test_api.py:24]: Added a dropout layer

File: tests/torch-ort_test_api.py (hunk 2 above)
Comment (thiagocrepaldi) [reply tests/torch-ort_test_api.py:33]: dropout was added. moving the input generation inside the loop makes the test fail

**Review by SherlockNoMad — APPROVED** [review]

File: tests/torch-ort_test_api.py (hunk 2 above)
Comment (SherlockNoMad) [reply tests/torch-ort_test_api.py:33]: yes, input generation is controlled by torch.manual_seed()
If you are interested, you can try reset  torch.manual_seed() in the loop

**Comment by thiagocrepaldi** [comment]
@natke @baijumeswani I need 2 approvals for merging this one :)

**Review by baijumeswani — APPROVED** [review]

**Review by natke — COMMENTED** [review]
Do we want to import it directly from onnxruntime, or add an interface to ORTModule?

**Comment by thiagocrepaldi** [comment]
> Do we want to import it directly from onnxruntime, or add an interface to ORTModule?

Setting seed is a global configuration, valid for the entire framework. Just like Pytorch's torch.nn.Module API doesn't have set seed while torch API has

In fact, this API is common to ORTTrainer and Inference too

**Review by liqunfu — APPROVED** [review]
